pkg/gofr/migration/mongo.go

```diff
@@ -0,0 +1,100 @@
+package migration
+
+import (
+	"context"
+	"time"
+
+	"gofr.dev/pkg/gofr/container"
+)
+
+type mongoDS struct {
+	container.Mongo
+}
+
+type mongoMigrator struct {
+	container.Mongo
+	migrator
+}
+
+// apply initializes mongoMigrator using the Mongo interface.
+func (ds mongoDS) apply(m migrator) migrator {
+	return mongoMigrator{
+		Mongo:    ds.Mongo,
+		migrator: m,
+	}
+}
+
+const (
+	mongoMigrationCollection = "gofr_migrations"
+)
+
+// checkAndCreateMigrationTable initializes a MongoDB collection if it doesn't exist.
+func (mg mongoMigrator) checkAndCreateMigrationTable(_ *container.Container) error {
+	err := mg.Mongo.CreateCollection(context.Background(), mongoMigrationCollection)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getLastMigration retrieves the latest migration version from MongoDB.
+func (mg mongoMigrator) getLastMigration(c *container.Container) int64 {
+	var lastMigration int64
+
+	var migrations []struct {
+		Version int64 `bson:"version"`
```

Comment:
Is it really a bson?


```suggestion
		Version int64 `json:"version"`
```

Reply:
As per my knowledge, we use bson while interacting with MongoDB , and json when using json parser as I am not marshalling unmarshalling anything, I think, I should use bson here.

Reply:
Thanks for checking